src/main/java/org/spongepowered/common/data/manipulator/mutable/item/SpongeHideData.java

```diff
@@ -0,0 +1,201 @@
+
+import com.google.common.collect.ComparisonChain;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.item.ImmutableHideData;
+import org.spongepowered.api.data.manipulator.mutable.item.HideData;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.manipulator.immutable.item.ImmutableSpongeHideData;
+import org.spongepowered.common.data.manipulator.mutable.common.AbstractData;
+import org.spongepowered.common.data.value.SpongeValueFactory;
+
+public class SpongeHideData extends AbstractData<HideData, ImmutableHideData> implements HideData {
+
+    private boolean enchantments;
+    private boolean attributes;
+    private boolean unbreakable;
+    private boolean canDestroy;
+    private boolean canPlace;
+    private boolean miscellaneous;
+
+    public SpongeHideData(boolean enchantments, boolean attributes, boolean unbreakable, boolean canDestroy, boolean canPlace,
+            boolean miscellaneous) {
+        super(HideData.class);
+
+        this.enchantments = enchantments;
+        this.attributes = attributes;
+        this.unbreakable = unbreakable;
+        this.canDestroy = canDestroy;
+        this.canPlace = canPlace;
+        this.miscellaneous = miscellaneous;
+
+        registerGettersAndSetters();
+    }
+
+    public SpongeHideData() {
+        this(false, false, false, false, false, false);
+    }
+
+    @Override
+    public Value<Boolean> hideEnchantments() {
+        return SpongeValueFactory.getInstance().createValue(Keys.HIDE_ENCHANTMENTS, this.enchantments, false);
+    }
+
+    @Override
+    public Value<Boolean> hideAttributes() {
+        return SpongeValueFactory.getInstance().createValue(Keys.HIDE_ATTRIBUTES, this.attributes, false);
+    }
+
+    @Override
+    public Value<Boolean> hideUnbreakable() {
+        return SpongeValueFactory.getInstance().createValue(Keys.HIDE_UNBREAKABLE, this.unbreakable, false);
+    }
+
+    @Override
+    public Value<Boolean> hideCanDestroy() {
+        return SpongeValueFactory.getInstance().createValue(Keys.HIDE_CAN_DESTROY, this.canDestroy, false);
+    }
+
+    @Override
+    public Value<Boolean> hideCanPlace() {
+        return SpongeValueFactory.getInstance().createValue(Keys.HIDE_CAN_PLACE, this.canPlace, false);
+    }
+
+    @Override
+    public Value<Boolean> hideMiscellaneous() {
+        return SpongeValueFactory.getInstance().createValue(Keys.HIDE_MISCELLANEOUS, this.miscellaneous, false);
+    }
+
+    @Override
+    public HideData copy() {
+        return new SpongeHideData(this.enchantments, this.attributes, this.unbreakable, this.canDestroy, this.canPlace, this.miscellaneous);
+    }
+
+    @Override
+    public ImmutableHideData asImmutable() {
+        return new ImmutableSpongeHideData(this.enchantments, this.attributes, this.unbreakable, this.canDestroy, this.canPlace, this.miscellaneous);
```

Comment:
Given that there's at most 36 possibilities, it's probably best to cache the immutable version.